src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Sdk.Before.targets

```diff
@@ -21,4 +23,41 @@
     <RuntimeIdentifier Condition="'$(UseRidGraph)' != 'true'">win-$(_SingleProjectHostArchitecture)</RuntimeIdentifier>
     <_MauiUsingDefaultRuntimeIdentifier>true</_MauiUsingDefaultRuntimeIdentifier>
   </PropertyGroup>
+  
+  <!--
+    Update the SupportedOSPlatformVersion and TargetPlatformMinVersion properties to
+    the .NET MAUI minimum supported versions if they are lower or missing.
+  -->
+  <PropertyGroup Condition="'$(TargetFramework)' != '' and '$(MauiDisableSupportedOSPlatformVersionValidation)' != 'true'">
+    <!-- Capture the current version information for logging. -->
+    <_MauiPreviousSupportedOSPlatformVersion>$(SupportedOSPlatformVersion)</_MauiPreviousSupportedOSPlatformVersion>
+    <_MauiPreviousTargetPlatformMinVersion>$(TargetPlatformMinVersion)</_MauiPreviousTargetPlatformMinVersion>
+    <!-- Windows has 2 properties that could be set, so if only 1 is set we copy it to the other. -->
+    <TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows' and '$(TargetPlatformMinVersion)' == ''">$(SupportedOSPlatformVersion)</TargetPlatformMinVersion>
+    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows' and '$(SupportedOSPlatformVersion)' == ''">$(TargetPlatformMinVersion)</SupportedOSPlatformVersion>
+    <!-- Record that the explicit version number was lower than we support so we can increase and log later. -->
+    <_MauiHadToIncreaseSupportedOSPlatformVersion Condition="'$(SupportedOSPlatformVersion)' != '' and $([MSBuild]::VersionGreaterThan('$(_MauiSupportedOSPlatformVersion)', '$(SupportedOSPlatformVersion)'))">true</_MauiHadToIncreaseSupportedOSPlatformVersion>
+    <_MauiHadToIncreaseTargetPlatformMinVersion Condition="'$(TargetPlatformMinVersion)' != '' and $([MSBuild]::VersionGreaterThan('$(_MauiSupportedOSPlatformVersion)', '$(TargetPlatformMinVersion)'))">true</_MauiHadToIncreaseTargetPlatformMinVersion>
+    <!-- Specify the version numbers, and override if it is too low. -->
+    <SupportedOSPlatformVersion Condition="'$(SupportedOSPlatformVersion)' == '' or '$(_MauiHadToIncreaseSupportedOSPlatformVersion)' == 'true'">$(_MauiSupportedOSPlatformVersion)</SupportedOSPlatformVersion>
```

Comment:
I don't understand the logic here, are we trying to increase `$(SupportedOSPlatformVersion)` _for them_ when it is also blank?

I would only set `$(SupportedOSPlatformVersion)` if it is blank, don't try to raise it automatically. Just emit warnings.

Reply:
That is correct, we want to override by default for best results since maui users don't like fiddling with things. Maybe we can change it to just a warning for RC 1 if people complain? @Redth 

But, either way I added a msbuild property `$(MauiDisableOverrideSupportedOSPlatformVersion)` that people can set to turn this off if they want their apps to not launch. This is a case of a Blazor user decides to target Android 21, but nothing will run since the APIs are 24+. Instead of a soft upgrade, it will now just crash on launch. But, at least they decided to enable crash mode.

Reply:
Can we remove the logic that tries to automatically increase the numbers? Let's change this to only add warnings.